src/index.ts

```diff
@@ -0,0 +1,449 @@
+	 * @param {*} [thisArg] Value to use as `this` when executing function
+	 * @returns {boolean}
+	 * @example collection.some(user => user.discriminator === '0000');
+	 */
+	public some(fn: (value: V, key: K, collection: this) => boolean, thisArg?: any): boolean {
+		if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
+		for (const [key, val] of this) {
+			if (fn(val, key, this)) return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Checks if all items passes a test. Identical in behavior to
+	 * [Array.every()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every).
+	 * @param {Function} fn Function used to test (should return a boolean)
+	 * @param {*} [thisArg] Value to use as `this` when executing function
+	 * @returns {boolean}
+	 * @example collection.every(user => !user.bot);
+	 */
+	public every(fn: (value: V, key: K, collection: this) => boolean, thisArg?: any): boolean {
+		if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
+		for (const [key, val] of this) {
+			if (!fn(val, key, this)) return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Applies a function to produce a single value. Identical in behavior to
+	 * [Array.reduce()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce).
+	 * @param {Function} fn Function used to reduce, taking four arguments; `accumulator`, `currentValue`, `currentKey`,
+	 * and `collection`
+	 * @param {*} [initialValue] Starting value for the accumulator
+	 * @returns {*}
+	 * @example collection.reduce((acc, guild) => acc + guild.memberCount, 0);
+	 */
+	public reduce<T extends any>(fn: (accumulator: any, value: V, key: K, collection: this) => T, initialValue?: T): T {
+		let accumulator!: T;
+
+		if (typeof initialValue !== 'undefined') {
+			accumulator = initialValue;
+			for (const [key, val] of this) accumulator = fn(accumulator, val, key, this);
+			return accumulator;
+		}
+		let first = true;
+		for (const [key, val] of this) {
+			if (first) {
+				accumulator = val as unknown as T;
+				first = false;
+				continue;
+			}
+			accumulator = fn(accumulator, val, key, this);
+		}
+
+		// No items iterated.
+		if (first) {
+			throw new TypeError('Reduce of empty collection with no initial value');
+		}
+
+		return accumulator;
+	}
+
+	/**
+	 * Identical to
+	 * [Map.forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach),
+	 * but returns the collection instead of undefined.
+	 * @param {Function} fn Function to execute for each element
+	 * @param {*} [thisArg] Value to use as `this` when executing function
+	 * @returns {Collection}
+	 * @example
+	 * collection
+	 *  .each(user => console.log(user.username))
+	 *  .filter(user => user.bot)
+	 *  .each(user => console.log(user.username));
+	 */
+	public each(fn: (value: V, key: K, collection: Map<K, V>) => void, thisArg?: any): this {
```

Comment:
Might be worth leaving the return type of `fn` as any so people can use arrow functions here.